plugin/omero_iviewer/views.py

```diff
@@ -555,12 +555,28 @@ def delta_t_data(request, image_id, conn=None, **kwargs):
     if size_t > 1:
         params = omero.sys.ParametersI()
         params.addLong('pid', image.getPixelsId())
-        z = 0
-        c = 0
         query = "from PlaneInfo as Info where"\
-            " Info.theZ=%s and Info.theC=%s and pixels.id=:pid" % (z, c)
+            " Info.theZ=0 and Info.theC=0 and pixels.id=:pid"
         info_list = conn.getQueryService().findAllByQuery(
             query, params, conn.SERVICE_OPTS)
+
+        if len(info_list) < size_t:
+            # C & Z dimensions are not always filled
+            # Remove restriction on c0 z0 to catch all timestamps
+            params = omero.sys.ParametersI()
+            params.addLong('pid', image.getPixelsId())
+            query = """
+                from PlaneInfo Info where Info.pixels.id=:pid
```

Comment:
it's harmless, but you don't need this `where Info.pixels.id=:pid` as you're selecting PlaneInfo by the same criteria below

Reply:
right, I remove it for simplicity.